utils/fwutils.py

```diff
@@ -127,6 +139,15 @@ def reflash_ipu(self) -> None:
         else:
             self.logger.info("Skipping flash_spi_image")
 
+        self.logger.info("Step 5: apply_fixboard")
+        if self.should_run("apply_fixboard"):
+            should_fixboard = self.fixboard_is_needed()
```

Comment:
Why make fixboard_is_needed() return an optional bool?

In my mind this could be simplified a bit.

```
if self.fixboard_is_needed():
    self.apply_fixboard()
else:
     # assume fixboard has already been applied, or the user does not want to apply this (though I'm not sure if the user has a knob to tell fwup to only run a subset of the steps anyways right now).
```

If we hit an exception when trying to determine if we should apply fixboard, something is wrong and we should bail out with an error code.

Reply:
The user does not have a knob yet for should_run but here it skips if it shouldn't run, then checks if it needs to use fixboard, and inside the function when it errors out it logs the exception and returns a None, so in the reflash step i assert that it isn't none , but i am changing it to check then exit with an error code instead

Reply:
Just added the changes, and tested it with a reflash, seems good to go!